packages/x-components/src/components/base-extra-params.vue

```diff
@@ -0,0 +1,165 @@
+<script lang="ts">
+  import Vue from 'vue';
+  import { Component, Prop, Watch } from 'vue-property-decorator';
+  import { Dictionary } from '../utils';
+  import { ExtraParamXEvent } from '../wiring';
+  import { SnippetConfig } from '../x-installer';
+  import { XInject } from './decorators/injection.decorators';
+
+  /**
+   * It emits a list of {@link ExtraParamXEvent} and the values, which are merged
+   * with the extra params from the snippet config, passed as properties.
+   *
+   * Each extra param is emitted by each event.
+   *
+   * By default if there is an extra param passed as prop, it will override
+   * the snippet config in case it exists.
+   *
+   * @public
+   */
+  @Component
+  export default class BaseExtraParams extends Vue {
```

Comment:
I'm going to put here what we talked yesterday:

* Create a new x-module (sorry for that 😅). We can do this in another PR.
* 3 events: `ExtraRequestParamsProvided`, and `UserChangedRequestParams` (this one you won't use it yet) emitted from the components, and `ExtraRequestParamsChanged`, emitted from a watcher in the store.
	* Modules that need the extra params are subscribed to `ExtraRequestParamsChanged` using the wiring.
	* `ExtraRequestParamProvided` and `UserChangedRequestParams` is only used in the extra params module.
* 2 components:
	* `<ExtraRequestParams :values="myExtraRequestParams" />`
	  * Where `myExtraRequestParams` is a dictionary
		* Uses `@XEmit` with `immediate` and `deep`
		* Does not have to perform diffs because `value` prop is trust to be different.
	* `<SnippetConfigExtraRequestParams />`
		* Renders an `ExtraRequestParams` component.
		* Keeps manually an object with the desired snippet config properties to avoid unnecessary re-renders:
```
    public extraRequestParams: Dictionary<unknown> = { };

    @Watch('snippetConfig', { deep: true, immediate: true })
    syncExtraRequestParams({
      lang,
      searchLang,
      instance,
      scope,
     // Every other param
      ...extraRequestParams
    }: SnippetConfig): void {
      forEach(extraRequestParams, (name, value) => {
        this.$set(this.extraRequestParams, name, value);
      });
    }
```

Reply:
Note that this is the initial idea, but when doing it we might realise that things make more sense in other way.